viewer-prototype/src/browser/trace-viewer/components/timegraph-output-component.tsx

```diff
@@ -100,24 +107,39 @@ export class TimegraphOutputComponent extends AbstractTreeOutputComponent<Timegr
             this.rowController.totalHeight = this.totalHeight;
             // TODO Style should not be retreive in the "initialization" part or at least async
             const styleResponse = (await this.props.tspClient.fetchStyles(this.props.traceId, this.props.outputDescriptor.id, QueryHelper.query())).getModel();
+            // TODO Remove filter when timeline chart can manage empty rows
+            const filteredTree = treeResponse.model.entries.filter((entry: TimeGraphEntry) => entry.parentId !== -1);
             this.setState({
                 // outputStatus: ResponseStatus.COMPLETED,
-                timegraphTree: treeResponse.model.entries,
+                timegraphTree: filteredTree,
                 styleModel: styleResponse.model
             });
+            const treeNodes = listToTree(filteredTree, 0);
+            const orderedTreeIds = getOrderedIds(treeNodes);
+            this.createChartLayer(orderedTreeIds);
+        }
+        if (this.state.collapsedNodes !== _prevState.collapsedNodes) {
+            this.updateChart();
         }
     }
 
+    private updateChart() {
+        // TODO: update timeline chart with visible rows (not collapsed)
+        // Get rows with tsp-data-provider using getAllVisibleEntriesId() in filter-tree/utils.tsx
+        // TODO: manage empty lines chart for root nodes with no data
+    }
+
     renderTree(): React.ReactNode {
-        return <React.Fragment>
-            {this.state.timegraphTree.map((entry, i) => {
-                if (entry.parentId !== -1) {
-                    return <p style={{ height: this.props.style.rowHeight, margin: 0 }} key={i}>
-                        {entry.labels[0] + '\n'}
-                    </p>;
-                }
-            })}
-        </React.Fragment>;
+        this.onCollapse = this.onCollapse.bind(this);
+        return  <EntryTree
+            collapsedNodes={this.state.collapsedNodes}
+            padding={12}
+            rootId={0}
```

Comment:
I don't think you can assume that the root ID is 0. The root ID could be any number. In method listToTree the root nodes should be auto discovered.